### DISCUSSION TITLE
Supabase realtime deletes and RLS

### DISCUSSION THREAD

**Discussion by aviadmini** [supabase/supabase#16631] [Questions]
I've set up RLS policies to allow users to insert, update and delete data associated with their user_id `USING (auth.uid() = user_id)`

When I subscribe to realtime postgres changes, inserts and updates work fine but I'm getting deletes from other users that aren't supposed to be visible to the user. 

The only reference to something similar I've found is this one here https://supabase.com/docs/guides/realtime/postgres-changes#full-old-record
![image](https://github.com/supabase/supabase/assets/5710176/aac48a5b-f713-4d50-97ed-3f6ea3b657bb)

Is there a way to prevent this?

**Comment by GaryAustin1** [comment]
Not currently.   It is why they only return the primary keys so you would not get other user's info from the other columns.

**Reply by aviadmini** [reply]
I suppose one could use soft deletes and get rid of the delete spam from the socket. With replica identity set to full the updates get the old record and it's possible to filter and process the soft deletes. Unfortunately, there's only one filter supported on the postgres changes subscription so it might still cause extra client work when you already use filters.

**Reply by GaryAustin1** [reply]
This may block filters being useable on deletes right now... https://github.com/supabase/walrus/issues/53
